core/common/scale.hpp

```diff
@@ -0,0 +1,390 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_SCALE_HPP
+#define KAGOME_SCALE_HPP
+
+#include <cstdint>
+#include <vector>
+
+#include <boost/multiprecision/cpp_int.hpp>
+
+#include "common/result.hpp"
+
+namespace kagome::common {
+
+  /**
+   * Scale namespace contains functions and classes
+   * for encoding and decoding values in and from SCALE encoding.
+   * There is no way to specify what exactly data is encoded,
+   * and specification has no mentions concerning this fact.
+   * However the SUBSTRATE documentation says directly, that
+   * SCALE encoding is not self-describing and that encoding context
+   * has to know what data it needs to decode.
+   * In this concern, signed and unsigned integers encoded the same way.
+   * When decoding you have to reinterpret integer value as signed
+   * if you know it is signed.
+   *
+   * This library provides following options:
+   * 1. encode/decode fixed width integers: 1,2,4 byte integers:
+   *    it only translates them from host to little-endian and vice-versa
+   * 2. encode/decode boolean values: true is encoded as 0x01,false as 0x00
+   * 3. encode/decode tristate values: none is 0x00, false is 0x01, true is 0x02
+   * 4. encode/decode one-byte options
+   * 5. collections of same type values are encoded/ decoded by following
+   * scheme: first value is compact-encoded number of items followed by each
+   * item scale-encoded sequencially
+   * 6. byte arrays are example of collections
+   * 7. enumerations
+   *
+   */
+
+  namespace scale {
+    using ByteArray = std::vector<uint8_t>;
+    using BigInteger = boost::multiprecision::cpp_int;
+
+    namespace fixedwidth {
+      /// Encodes Plain integer of any byte size signed or unsigned from host-
+      /// to little- endian
+      /// \tparam T Integer type
+      /// \param value Source value
+      /// \return Encoded value
+      template <class T>
+      T encodePlainInteger(T value);
```

Comment:
maybe it can be useful: just encodePlainInteger()
for every value you have if you are sure they have suitable type
or maybe use encodeUint**() if you are not.
Previously I was told to make those methods, so I did.

Reply:
agree that it better remove it from api

Reply:
I listen and obey...)